spinalcordtoolbox/scripts/sct_process_segmentation.py

```diff
@@ -397,6 +410,8 @@ def main(argv: Sequence[str]):
     qc_dataset = arguments.qc_dataset
     qc_subject = arguments.qc_subject
 
+    if normalize_pam50 and not perslice:
+        parser.error("Option '-normalize-PAM50' requires option '-perslice 1'.")
```

Comment:
Just to double-check: Why is `-perslice 1` mandatory?

Out of curiosity, I tried removing this error, then specifying `-perlevel 1` instead of `-perslice 1`, and the results seem okay?

```
sct_process_segmentation -i t2_seg.nii.gz -vertfile t2_seg_labeled.nii.gz -perlevel 1 -normalize-PAM50 1 -o csa_pam50.csv
```

![image](https://user-images.githubusercontent.com/16181459/217014427-d5375f8c-c410-4f93-a65b-9ce796fcf2eb.png)

Is it worth considering removing hard requirement for `-perslice 1`?

Reply:
No, we want afterwards to do slicewise comparisons. The idea of puting everything in the PAM50 anatomical space is that we will have a slice correspondance between subjects! If we average across levels, using the pam50 anatomical dimensions is not useful, we could compare perlevel in the native space .

Reply:
That's very fair! I can see why it's only really useful to use `-normalize-PAM50` with `-perlevel 1`.

Still... it isn't exactly a _mandatory_ step? (It's not like `-normalize-PAM50` will fail without `-perlevel 1`, because the normalization step is completely separate from the aggregation step.) 

Technically, saying that it is a hard requirement is a bit misleading... instead, `-perlevel 1` is more like a strong recommendation?. Because of this, I feel like it may be better to remove the strict requirement, and instead explain the `-perlevel 1` recommendation in the usage:

```diff
    optional.add_argument(
        '-normalize-PAM50',
        metavar=Metavar.int,
        type=int,
        choices=[0, 1],
        default=0,
-       help="Set to 1 to bring the metrics in the PAM50 anatomical dimensions perslice. -vertfile and -perslice need to be specified."
+       help="Set to 1 to bring the metrics in the PAM50 anatomical dimensions. -vertfile must be specified.\n"
+            "NB: It is strongly recommended to always use this option in combination with '-perslice 1', as this will enable you to "
+            "perform slicewise comparisons of normalized metrics between different subjects."
    )
```

That way, the user needs think about the aggregation setting that they are choosing, and will hopefully have a better understanding of _why_ they should be setting `-perlevel 1`.

Reply:
I am not sure I would want to remove the mandatory `-perslice 1` since the option` -normalize-PAM50` is kind of useless without... Maybe @jcohenadad and @valosekj you have an opinion on this?

Reply:
I agree with Sandrine. The option `-normalize-PAM50` is designed to bring morphometric metrics from the subject native space to the PAM50. This normalization is done using a linear interpolation of [individual slices per level](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/5722cab228cd976b91d3b6f1a6c2a3ba2f2f6a78/spinalcordtoolbox/metrics_to_PAM50.py#L10-L12). This means that we need individual slices (the option `-perslice 1`) for each level. Without the option `-perslice 1`, we would have only a single value for each level, and interpolation would not be possible.

Reply:
> This means that we need individual slices (the option `-perslice 1`) for each level. Without the option `-perslice 1`, we would have only a single value for each level, and interpolation would not be possible.

Just to clarify one thing: the PAM50 normalization step occurs before any aggregation steps. So, regardless of `-perslice 1` the normalization will be performed with the full per-slice metrics.

Whether or not we specify `-perslice 1` only affects how the per-slice metrics then get aggregated for the output CSV file.

Reply:
> > This means that we need individual slices (the option `-perslice 1`) for each level. Without the option `-perslice 1`, we would have only a single value for each level, and interpolation would not be possible.
> 
> Just to clarify one thing: the PAM50 normalization step occurs before any aggregation steps. So, the setting of `-perslice` has no effect on the normalization/interpolation steps; they will be performed with the full per-slice metrics regardless of how `-perslice` is set..
> The option `-perslice 1` only affects how the per-slice metrics get aggregated for the output CSV file. (So, the discussion is whether we should make it mandatory for the output CSV to be per-slice, or whether it should be explained in the usage only.)

I see! Now I get what you mean! Exactly; metrics are: 1) computed slice-wise by [`compute_shape`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/5722cab228cd976b91d3b6f1a6c2a3ba2f2f6a78/spinalcordtoolbox/scripts/sct_process_segmentation.py#L423), 2) interpolated by [`interpolate_metrics`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/5722cab228cd976b91d3b6f1a6c2a3ba2f2f6a78/spinalcordtoolbox/scripts/sct_process_segmentation.py#L430), and 3) aggregated using [`aggregate_per_slice_or_level`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/5722cab228cd976b91d3b6f1a6c2a3ba2f2f6a78/spinalcordtoolbox/scripts/sct_process_segmentation.py#L463). 
So, yes, as you mentioned, technically, `-normalize-PAM50` works also with `-perlevel 1` (instead of `-perslice 1`) because interpolation/normalization is done slice-wise regardless of how `-perslice` is set.

Reply:
yes, it is correct in the sens of the interpolation, however, it would give the same results if it is `-perlevel 1` than if we don't use `-normalize-PAM50` since it is averaged on the entire level. The only thing that would differ are the slices numbers. This is what I mean when it is useless. I will discuss in our meeting with Julien tomorrow and get back to you @joshuacwnewton !

Reply:
Ahhhh, I see!!! In that case, I think it's probably fine to keep things as they are!